### DISCUSSION TITLE
Keycloak 16 Enabling offline access in client scope

### DISCUSSION THREAD

**Discussion by TharinduGimhana** [keycloak/keycloak#24208] [Q&A]
In my project, we are using keycloak (16.1.1) as IDP. We are facing an issue when the server restarted all the sessions will be removed. to resolve this I'm trying to generate tokens with passing scope as scope = offline_access.

1. I assigned offline access to the client scopes as below

![image](https://github.com/keycloak/keycloak/assets/28914539/6179adbe-be9e-4c00-9616-a3fba26c95cb)

2. I tried to generate a client token with passing scope as scope = offline_access and I got an error below
{
    "error": "not_allowed",
    "error_description": "Offline tokens not allowed for the user or client"
}

![image](https://github.com/keycloak/keycloak/assets/28914539/dbfd2102-4fd8-4632-98ac-e3490ca40de4)

4. I checked my error logs from the server and I got the error below
**KC-SERVICES0060: Role 'offline_access' not available in the realm**

I'm looking for advice, on how to add the Offline access in client scope for keycloak 16.1.1.

Thank you.

**Comment by danielFesenmeyer** [comment]
Does the user have the offline_access role? https://www.keycloak.org/docs/latest/server_admin/#_offline-access

**Reply by danielFesenmeyer** [reply]
Ah, now I saw the error message 4. It's weird that the offline_access role doesn't exist. I've never seen this.
